### PR TITLE
Try to fix CJK search

### PR DESCRIPTION
**Pull request by Ovler-Young**
该修改将不再替换搜索词中除数字字母以外的其他字符，将大大改善搜索体验。
另：该api以文件内容搜索，还是较为准确的。

### CODE REVIEW

**Comment by Ovler-Young** [comment]
Before
修改前

---
After
修改后

**Comment by myl7** [comment]
API 跑了几下感觉确实还行？OneDrive API 开窍了？

File: pages/api/search.ts [hunk 1]
```diff
@@ -12,7 +12,10 @@ import siteConfig from '../../config/site.config'
  * @returns Sanitised query string which replaces non-alphanumeric characters with ' '
  */
 function sanitiseQuery(query: string): string {
-  const sanitisedQuery = query.replace(/[^a-zA-Z0-9]/g, ' ')
+  const sanitisedQuery = query
+/** Actually this will cause problems with CJK only query words as it will remove every character.
+ * After removing it, CJK search will be much improved.
+ */
```
Comment (spencerwooo) [inline pages/api/search.ts:18]: Delete these comments and move them up to the function descriptions

File: pages/api/search.ts [hunk 2]
```diff
@@ -12,7 +12,10 @@ import siteConfig from '../../config/site.config'
  * @returns Sanitised query string which replaces non-alphanumeric characters with ' '
  */
 function sanitiseQuery(query: string): string {
-  const sanitisedQuery = query.replace(/[^a-zA-Z0-9]/g, ' ')
+  const sanitisedQuery = query
```
Comment (spencerwooo) [inline pages/api/search.ts:15]: You are still not removing characters including `'`, which causes 400 errors. Take care of this issue.

**Review by spencerwooo — CHANGES_REQUESTED** [review]

File: pages/api/search.ts (hunk 2 above)
Comment (spencerwooo) [reply pages/api/search.ts:15]: FYI: https://github.com/spencerwooo/onedrive-vercel-index/issues/302

**Review by spencerwooo — APPROVED** [review]
🚀

**Pull request merged**